core/src/banking_trace.rs

```diff
@@ -0,0 +1,497 @@
+            if err.kind() == io::ErrorKind::NotFound {
+                Ok(())
+            } else {
+                Err(err)
+            }
+        })
+    }
+
+    fn create_file_appender(
+        path: PathBuf,
+        rotate_threshold_size: u64,
+    ) -> Result<RollingFileAppender<RollingConditionGrouped>, TraceError> {
+        let grouped = RollingConditionGrouped::new(
+            RollingConditionBasic::new()
+                .daily()
+                .max_size(rotate_threshold_size),
+        );
+        let appender = RollingFileAppender::new_with_buffer_capacity(
+            path.join(BASENAME),
+            grouped,
+            (TRACE_FILE_ROTATE_COUNT - 1).try_into()?,
+            BUF_WRITER_CAPACITY,
+        )?;
+        Ok(appender)
+    }
+
+    fn spawn_background_thread(
+        trace_receiver: Receiver<TimedTracedEvent>,
+        mut file_appender: RollingFileAppender<RollingConditionGrouped>,
+        exit: Arc<AtomicBool>,
+    ) -> Result<JoinHandle<TracerThreadResult>, TraceError> {
+        let thread = thread::Builder::new().name("solBanknTracer".into()).spawn(
+            move || -> TracerThreadResult {
+                sender_overhead_minimized_receiver_loop::<
+                    _,
+                    TraceError,
+                    TRACE_FILE_WRITE_INTERVAL_MS,
+                >(exit, trace_receiver, |event| -> Result<(), TraceError> {
+                    file_appender.condition_mut().reset();
+                    serialize_into(&mut GroupedWriter::new(&mut file_appender), &event)?;
+                    Ok(())
+                })?;
+                file_appender.flush()?;
+                Ok(())
+            },
+        )?;
+
+        Ok(thread)
+    }
+}
+
+pub struct TracedSender {
+    label: ChannelLabel,
+    sender: Sender<BankingPacketBatch>,
+    trace_sender: Option<(Sender<TimedTracedEvent>, Arc<AtomicBool>)>,
+}
+
+impl TracedSender {
+    fn new(
+        label: ChannelLabel,
+        sender: Sender<BankingPacketBatch>,
+        trace_sender: Option<(Sender<TimedTracedEvent>, Arc<AtomicBool>)>,
+    ) -> Self {
+        Self {
+            label,
+            sender,
+            trace_sender,
+        }
+    }
+
+    pub fn send(&self, batch: BankingPacketBatch) -> Result<(), SendError<BankingPacketBatch>> {
+        if let Some((trace_sender, exit)) = &self.trace_sender {
+            if !exit.load(Ordering::Relaxed) {
+                trace_sender
+                    .send(TimedTracedEvent(
+                        SystemTime::now(),
+                        TracedEvent::PacketBatch(self.label, Arc::clone(&batch)),
+                    ))
+                    .expect("active tracer thread unless exited");
+            }
+        }
+        self.sender.send(batch)
+    }
+}
+
+pub fn sample_packet_batch() -> BankingPacketBatch {
+    BankingPacketBatch::new((to_packet_batches(&vec![test_tx(); 4], 10), None))
+}
+
+pub fn drop_and_clean_temp_dir_unless_suppressed(temp_dir: TempDir) {
```

Comment:
Looks like these fns are only used by tests and benches; might be obvious to others since they are outside the class, but a comment above this group of fns could help

Reply:
done (with a bit of twist; i hate maintaining comments and see stale ones...lol): 6a863d685fb62d45f6304f97ca406f8199baffd5

Reply:
Ah perfect; I actually prefer a separate module for testing/bench utils, thought it might be overkill to ask!